[Settings] Update to Calypso Site Admin package

psealock opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

<!-- For bug fixes: If known, please provide links to help with traceability and escape analysis. -->
<!-- Please include a link to the issue of the bug being fixed, if one doesn't exist please create it. -->
<!-- If the PR that introduced the bug is known, please also add its link below. -->

Closes https://github.com/woocommerce/woocommerce/issues/53818

This PR introduces shared a8c components created in https://github.com/Automattic/wp-calypso/pull/99754, starting with just the Router.

The aim of this PR is to swap out the previous router from `@wordpress/edit-site` to `@calyspso/site-admin`. I wanted to use the new router without making any major refactors in the way data and routing is handled. You can see the routes passed to `RouterProvider` is just an empty array (for now), `<RouterProvider routes={ [] } pathArg="page">`.

### Notable changes

* I removed directly calling `possiblyRenderSettingsSlots` and related slot registrations. It turns out the entirety of Settings slots have been condensed to a single external script [wp-admin-scripts/settings-embed](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/client/admin/client/wp-admin-scripts/settings-embed/index.tsx). This external script gets executed just like any other script, including ones any extension may use to render slotFill. So by making this script work, we in turn assure 3PD scripts also work.
* I greatly simplified the root of the application, `plugins/woocommerce/client/admin/client/settings/index.js`, which previously was responsible for handling scripts. In order to keep the router and all its routes within the Settings Editor package, I needed to move the append scripts logic to the settings package, where it makes more sense anyways.
* There was a lot of noise in the console from plugins being re-registered. I added a check before plugins register to avoid re-registering. Because of this, testing with the `settings` feature flag is required.
* I moved `settingsScripts` into the data context.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. ~Add/Enable Gutenberg 19.8 or earlier.~ Since https://github.com/woocommerce/woocommerce/pull/56271 has been merged, there is no longer a Gutenberg dependency.
2. Before turning on the feature flag for `settings`, poke around the current pages. Ensure pages with slotFill render normally, such as Site Visibility, Email, or Payments.
3. Turn on the `settings` feature flag.
4. Review the same settings screens and see the slotFills rendered. Navigate around the app and ensure content refreshes appropriately.
5. Install the [Migration Tester](https://github.com/woocommerce/woocommerce-settings-migration-tester) by running `npm run zip` from project root and installing the zip.
6. Test the various tabs in Legacy Settings and Modern Settings - they should function normally. Note Legacy Settings > Javascript isn't highlighting properly. I'd like to address this in a follow up because the same issue can be seen on `trunk`.

Interesting to note the stacked modal issue reported in [[Settings] Implement proper script injection and state management](https://github.com/woocommerce/woocommerce/issues/55531) is no longer visible. I don't think that issue is solved, but something has changed.

<!-- End testing instructions -->

### Testing that has already taken place:

<!-- Detail any testing that has already been conducted. -->
<!-- Include environment details such as hosting type, plugins, theme, store size, store age, and relevant settings. -->
<!-- Mention any analysis performed, such as assessing potential impacts on environment attributes and other plugins, performance profiling, or LLM/AI-based analysis. -->
<!-- Within the testing details you provide, please ensure that no sensitive information (such as API keys, passwords, user data, etc.) is included in this public pull request. -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [x] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

Update new Settings to use @automattic/site-admin's router instead of Gutenberg

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

psealock committed:
use location to get route

psealock committed:
get links working

psealock committed:
move script creation to inside settingsApp

psealock committed:
suppress registered plugin warnings

psealock committed:
use settings-embed

psealock committed:
external slotFill working

psealock committed:
remove console.log

psealock committed:
cleanup

psealock committed:
Merge branch 'trunk' into update/settings-router-calypso-package

psealock committed:
remove unused Location

psealock committed:
get site admin package from npm

psealock committed:
Merge branch 'trunk' into update/settings-router-calypso-package

Commit:
Add changefile(s) from automation for the following project(s): @woocommerce/settings-editor, woocommerce, woocommerce/client/admin

psealock committed:
Merge branch 'trunk' into update/settings-router-calypso-package

psealock committed:
lint

psealock committed:
fix tests

kalessil commented on packages/js/settings-editor/package.json:
```diff
@@ -34,6 +34,7 @@
 		"access": "public"
 	},
 	"dependencies": {
+		"@automattic/site-admin": "^0.0.1",
```
It looks like a new dependency, we should add it to `.syncpackrc` to ensure the monorepo will use same version of this dependency.

kalessil replied:
Thanks!

psealock committed:
add site admin syncpack entry

psealock committed:
Merge branch 'trunk' into update/settings-router-calypso-package

psealock committed:
fix snackbar dismiss

rjchow reviewed (approved):
An inspection of the code in this PR looks ok to me, and I've tested the functionality of this PR when testing https://github.com/woocommerce/woocommerce/pull/56271 so I'm omitting testing here.

psealock committed:
Merge branch 'trunk' into update/settings-router-calypso-package

psealock committed:
lockfile

psealock committed:
lockfile, again

psealock committed:
lockfile

psealock committed:
[Settings] Update sidebar components calypso package (#56271)

psealock committed:
Merge branch 'trunk' into update/settings-router-calypso-package

psealock committed:
fix syncpack

psealock committed:
lint

psealock committed:
remove unused private apis dependency

psealock committed:
sync all private-apis to wp-6.6

psealock committed:
reset .syncpackrc

Commit:
Add changefile(s) from automation for the following project(s): @woocommerce/settings-editor, @woocommerce/product-editor, woocommerce, woocommerce/client/admin

psealock committed:
Merge branch 'trunk' into update/settings-router-calypso-package

chihsuan committed:
Fix tests

- Updated `@wordpress/private-apis` version from `wp-6.6` to `wp-6.7` in `package.json`.
- Adjusted Jest configuration to ensure consistent usage of WP packages across tests.
- Refactored test snapshots for `SearchListControl` to align with updated component structure.
- Removed deprecated `@types/wordpress__components` references from `package.json`.
- Added mappings for WP packages in the Jest preset to prevent version conflicts during testing.

chihsuan committed:
Merge branch 'trunk' into update/settings-router-calypso-package

Commit:
Add changefile(s) from automation for the following project(s): @woocommerce/settings-editor, @woocommerce/product-editor, @woocommerce/internal-js-tests, woocommerce, woocommerce/client/admin

chihsuan committed:
Fix tests

chihsuan committed:
Add changelog

chihsuan commented on plugins/woocommerce/src/Admin/Features/Settings/Init.php:
```diff
@@ -41,6 +41,7 @@ public function __construct() {
 		add_filter( 'woocommerce_admin_shared_settings', array( __CLASS__, 'add_component_settings' ) );
 		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_settings_editor_scripts' ) );
 		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_settings_editor_styles' ) );
+		add_action( 'woocommerce_navigation_is_connected_page', array( $this, 'disable_embed_if_woocommerce_settings_page' ) );
```
I found that the `payment-method-promotions` style is not being loaded because we disabled the embed page. This caused some styles to be missing.

Trunk:
![Screenshot 2025-03-25 at 15 33 13](https://github.com/user-attachments/assets/26e5805c-0a56-4cd8-aeea-bf8def2c6722)

This branch:

![Screenshot 2025-03-25 at 15 32 08](https://github.com/user-attachments/assets/f6c97841-9e90-40c4-928a-6056361b8d8e)

psealock replied:
Good find @chihsuan, thank you.

I think excluding the new Settings as an `embed` page was too heavy handed. Many styles, including some from your screenshot, are actually included in [styles loaded based on the page being an embed](https://github.com/woocommerce/woocommerce/blob/f56bdb35838ab5f13c7e5067f0fb5d1c32730ae7/plugins/woocommerce/src/Internal/Admin/WCAdminAssets.php#L249). Instead, I returned Settings as an embed page, created a `PageController::is_modern_settings_page` function and conditionally loaded the embed header.

Fixed in [Return Settings to an 'embed' page](https://github.com/woocommerce/woocommerce/pull/56246/commits/3086594919daf1737f71a1e27954b28cd55b492a)

As far as the `payment-method-promotions` styles, they are loaded in a `admin_enqueue_scripts` hook. These don't fire when Settings loops through pages to gather assets and configs. I made a separate [issue to track this](https://github.com/woocommerce/woocommerce/issues/56737).

chihsuan committed:
Fix lint

chihsuan reviewed (commented):
Nice work! @psealock 👍 

Overall, this looks good. Just found an issue that styles are not being loaded because we disabled the embed page as commented in the PR.

I've updated the PR to fix unit test issues by aligning specific WordPress packages with their project counterparts for the test environment, as seen in commit 3b8085e320b76dcb32a3a52599cce9f4997188df. Additionally, I updated @woocommerce/blocks to ise `@wordpress/private-apis` version wp-6.7, as it is required by @wordpress/editor wp-6.7, which is incompatible with `@wordpress/private-apis` version wp-6.6.

psealock committed:
lockfile

psealock committed:
Use existing settings-embed loading

psealock committed:
Return Settings to an 'embed' page

psealock committed:
rest enqueue_assets

psealock committed:
whitespace

psealock committed:
Merge branch 'trunk' into update/settings-router-calypso-package

psealock committed:
embed styles

chihsuan reviewed (approved):
Thanks @psealock, good work on migrating to the Calypso Site Admin package! 👍 

I confirmed that styles are loaded correctly. While I also found a few issues such as can't save settings but I was able to reproduce them in trunk as well.
__

#### Test environment

<details>
- WordPress: 6.7.2
- Theme: Twenty Twenty-Five Version: 1.1
- WooCommerce - Development version with update/settings-router-calypso-package branch
- PHP: 8.3
- Web Server: Nginx
- Browser: Brave
- OS: macOS Sequoia 15.3.1
</details>

#### Review checklist

| Parameter  | Result |
| ------------- | ------------- |
| Approach  | ✅ Good migration strategy with backward compatibility |
| Implementation  | ✅ All good |
| Automated tests  | ✅ Tests updated appropriately |
| Naming  | ✅ All good |
| Typing  | ✅ All good |
| Code comments  | ✅ All good |
| Changelog  | ✅ Appropriate changelog entry |
| Documentation  | ✅ All good |
| PR description  | ✅ Clear and detailed |
| Testing instructions  | ✅ Comprehensive |
| Manual testing  | ✅ |

#### Scenarios tested

| Scenario  | Result | Screenshot |
| ------------- | ------------- | ------------- |
| Settings pages with slotFill | ✅ Function normally | |
| Legacy Settings tabs | ✅ Function normally | |
| Modern Settings tabs | ✅ Function normally | |
| Migration Tester | ✅ Works as expected | |

<img width="1147" alt="Screenshot 2025-03-27 at 10 48 37" src="https://github.com/user-attachments/assets/74153bfd-5bb3-49db-9e64-186bee25448a" />

<img width="1146" alt="Screenshot 2025-03-27 at 10 48 54" src="https://github.com/user-attachments/assets/52d4d3f7-8237-481a-863a-1159292f70b5" />

<img width="1147" alt="Screenshot 2025-03-27 at 10 49 03" src="https://github.com/user-attachments/assets/28863f34-6c0c-44ac-9796-236a897828c4" />

<img width="1147" alt="Screenshot 2025-03-27 at 10 49 21" src="https://github.com/user-attachments/assets/a48222e8-66ae-4f08-b22b-2f200a10fc5c" />
<img width="1147" alt="Screenshot 2025-03-27 at 10 49 49" src="https://github.com/user-attachments/assets/3f387458-6384-4a33-9bb0-e78a571226ac" />
